From aca2df64cd0cb6c4a8c03dc564c59f2a805dda7c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 4 Mar 2021 14:37:06 -0800
Subject: [PATCH] Fix build with gcc-11

* squash 2 patches from meta-qt5/master:
qmldebug: Include limits header

this is needed to solve
qmldebug/qqmlprofilerevent_p.h:314:65: error: 'numeric_limits' is not a member of 'std'

Its exposed bu gcc11

yarr: Include <limits> for numeric_limits

Fixes
src/3rdparty/masm/yarr/Yarr.h:46:44: error: 'numeric_limits' is not a member of 'std'
   46 | static const unsigned offsetNoMatch = std::numeric_limits<unsigned>::max();
      |                                            ^~~~~~~~~~~~~~

Upstream-Status: Pending

And add <limits> also to src/qml/jsruntime/qv4propertykey_p.h.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/3rdparty/masm/yarr/Yarr.h        | 2 +-
 src/qml/jsruntime/qv4propertykey_p.h | 1 +
 src/qmldebug/qqmlprofilerevent_p.h   | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/3rdparty/masm/yarr/Yarr.h b/src/3rdparty/masm/yarr/Yarr.h
index ccf78f9880..cbb42c60d8 100644
--- a/src/3rdparty/masm/yarr/Yarr.h
+++ b/src/3rdparty/masm/yarr/Yarr.h
@@ -27,7 +27,7 @@
 
 #pragma once
 
-#include <limits.h>
+#include <limits>
 #include "YarrErrorCode.h"
 
 namespace JSC { namespace Yarr {
diff --git a/src/qml/jsruntime/qv4propertykey_p.h b/src/qml/jsruntime/qv4propertykey_p.h
index 47867765db..c945b27498 100644
--- a/src/qml/jsruntime/qv4propertykey_p.h
+++ b/src/qml/jsruntime/qv4propertykey_p.h
@@ -50,6 +50,7 @@
 // We mean it.
 //
 
+#include <limits>
 #include <private/qv4global_p.h>
 
 QT_BEGIN_NAMESPACE
diff --git a/src/qmldebug/qqmlprofilerevent_p.h b/src/qmldebug/qqmlprofilerevent_p.h
index 1e205d8dbb..f2467bb032 100644
--- a/src/qmldebug/qqmlprofilerevent_p.h
+++ b/src/qmldebug/qqmlprofilerevent_p.h
@@ -47,6 +47,7 @@
 #include <QtCore/qvarlengtharray.h>
 #include <QtCore/qmetatype.h>
 
+#include <limits>
 #include <initializer_list>
 #include <type_traits>
 
